### PR TITLE
Add JS Analytics API

### PR DESCRIPTION
**Pull request by milkypostman**
Creates a class that can be used to send analytics data to app_engine
from the client.

### CODE REVIEW

File: src/web_app/js/analytics.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ */
+Analytics.prototype.sendEventRequest_ = function(eventObj) {
+  var request = {};
+  request[enums.RequestField.TYPE] = enums.RequestField.MessageType.EVENT;
+  request[enums.RequestField.REQUEST_TIME_MS] = Date.now();
+  request[enums.RequestField.EVENT] = eventObj;
+
+  sendAsyncUrlRequest('POST', this.analyticsPath_,
+      JSON.stringify(request)).then(function() {
+        trace('Successfully sent event.');
```
Comment (jiayliu) [inline src/web_app/js/analytics.js]: Will this spam the log if we send stats periodically?

Comment (milkypostman) [inline src/web_app/js/analytics.js]: yeah. I'll take it out.

**Comment by milkypostman** [comment]
Trace events removed. Waiting on merge of #133

**Comment by jiayliu** [comment]
lgtm

**Pull request merged**